PHP 8: clean up `Email` package

distantnative opened the pull request.

distantnative committed:
PHP 8: clean up `Email` package

lukasbestle commented on src/Email/Email.php:
```diff
@@ -123,8 +65,6 @@ public function __construct(array $props = [], bool $debug = false)
-	 *
-	 * @return array
 	 */
 	public function attachments(): array
```
All of these getter methods except `isSent()` can currently also return `null` because the props are all nullable.

Maybe we clean the prop types a bit by updating the setters to replace `null` with an empty array in the `setAttachments()`, `setBcc()`, `setCc()`, `setTo()` and `setTransport()` methods. The getters of the remaining props need to be updated to return nullable values.

distantnative replied:
Can they? `setAttachments` already ensure that an empty array is set if `null` is passed. Calling `resolveEmail()` in the setters also does that in the setters for bcc, cc, too. `::transport()` returns `::defaultTransport()` when it's null.

Type hinting at the top needs to include null as otherwise PHP throws an error, which I think is due to our `Properties` trait magic - not that the properties could actually be null once the constructor and magic has finished.

lukasbestle replied:
You are right. Then I think we should actually add `@var` comments to the props again to mark the actual possible types after the constructor has run. Otherwise we will be confused down the line.

distantnative replied:
Not sure if this would be less confusing (once you forget which one is the right one). I think the actual path forward is getting rid of the Properties trait.

distantnative replied:
Reconsidered, `@var` comments might be good. Adding them.

distantnative committed:
`Email\Email`: add more specific `@var` comments

Our `Props` trait requires many properties to be natively type hinted as alos `|null` - which isn’t really acurate. The `@var` comments clarify the actual types.

bastianallgeier commented on src/Email/Body.php:
```diff
@@ -19,20 +19,11 @@ class Body
-	 *
-	 * @param array $props
 	 */
 	public function __construct(array $props = [])
```
Wouldn't it be a good chance to refactor the constructor and get rid of the Properties trait, setText and setBody in one go? 

```php 
public function __construct(array $props = []) 
{
  $this->html = $props['html'] ?? null,
  $this->text = $props['text'] ?? null,
}
```

bastianallgeier reviewed (commented):
I think we could really get rid of all the protected set methods in Body and Email. I just don't know if this would lead to any potential breaking changes. Then it might not be worth it.
